### ISSUE TITLE
funclatency yielding odd results

### ISSUE THREAD

**Issue by Algodev-github** [iovisor/bcc#2219]
Hi,
funclatency is giving me bloated numbers, so I tried with an empty function, and funclatency tells me that this function lasts 1200 nsec, which is rather absurd on my 3.50 GHz CPU. Here are all details:
- I'm using a very recent version of bcc, cloned about ten days ago;
- the functions I'm measuring are invoked through pointers (if useful for you to know, I'm measuring the duration of the interface function of my BFQ I/O scheduler)
- to get stable results, I do the following:
        cpupower frequency-set -g performance -d 3.50GHz -u 3.50GHz
        cpupower idle-set -D 0

        (/usr/share/bcc/tools/funclatency $FUNC_TO_MEASURE) &
        BPFPID=$! # save pid, to stop funclatency when the test finishes

        # do the test during which I want to measure the duration of my function of interest

        kill -SIGINT $BPFPID
        cpupower frequency-set -g powersave -d 800MHz
        cpupower idle-set -E

Just ask if other information is needed.

**Comment by brendangregg** [comment]
> funclatency is giving me bloated numbers

by how much are they bloated? 1200ns? I'd subtract 1200ns from your results, since you just tested that the various BPF instructions to store and retrieve timestamps by tid take 1200ns on your system.

I would say that funclatency shouldn't show times < 1us, since it's likely unable to measure them on current processors, but that assumption won't stand the test of time.

**Comment by Algodev-github** [comment]
Yep, the functions I want to measure last less than 1us, so I guess I'll have to resort to direct code instrumentation.

Thanks!
